### PR TITLE
Integrate BittensorLLM

### PR DESCRIPTION
**Pull request by BitVoyagerMan**
Add NIBittensorLLM endpoint based on power of Bittensor

### CODE REVIEW

File: langchain/src/llms/bittensor.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+   Args:
+       prompt: The prompt to pass into the model.
+
+       Returns: The string generated by the model.
+
+   Example:
+   let response = niBittensorLLM.call("Tell me a joke.");
+   */
+  async _call(prompt: string): Promise<string> {
+    const defaultPrompt =
```
Comment (jacoblee93) [inline langchain/src/llms/bittensor.ts:57]: Why are these hard coded?

File: langchain/src/llms/bittensor.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      "You are an assistant which is created by Neural Internet(NI) in decentralized network named as a Bittensor.";
+
+    const systemPrompt: string =
+      this.systemPrompt === null || this.systemPrompt === undefined
+        ? `${defaultPrompt} Your task is to provide accurate response based on user prompt`
+        : `${defaultPrompt} ${this.systemPrompt}`;
+
+    try {
+      // Retrieve API KEY
+      const apiKeyResponse: Response = await fetch(
```
Comment (jacoblee93) [inline langchain/src/llms/bittensor.ts:67]: We're retrieving an API key from an unauthenticated endpoint?

**Comment by jacoblee93** [comment]
Can you add a docs page as well?

File: langchain/src/llms/bittensor.ts (hunk 1 above)
Comment (jacoblee93) [reply langchain/src/llms/bittensor.ts:57]: I mean why are we hard-coding prompts here? Are they somehow required?

Comment (BitVoyagerMan) [reply langchain/src/llms/bittensor.ts]: I fixed that hard coded prompts.
Thanks.

**Comment by jacoblee93** [comment]
There is no way that this is secure:

https://test.neuralinternet.ai/admin/api-keys/
```
[{"id":5,"api_key":"QBRYlXLdQ3A3oGkB4wutXIwXjscIbRd-zJCTWECvNpOp2pljTr02qLVEBeWUE7n0","api_key_hint":"...E7n0","name":"New API Key","request_count":14,"valid_until":-1,"credits":-1,"enabled":true,"rate_limits":null,"rate_limits_enabled":false,"created_at":1693507561,"updated_at":1693592070},{"id":4,"api_key":"gJ1naMFrz1xvGpyIWhD_LxK30RycK1MMG2SteS3vTlPcbwnJ91OO01OkkxtOq9Bz","api_key_hint":"...q9Bz","name":"New API Key","request_count":46,"valid_until":-1,"credits":-1,"enabled":true,"rate_limits":null,"rate_limits_enabled":false,"created_at":1693484437,"updated_at":1693504708},{"id":3,"api_key":"A1vjppoKLmoX4FfJAIW_CCsQG1rL7_24Hf0Pi9wSFBlmWcnAazpC_VSpKUXTHniC","api_key_hint":"...HniC","name":"New API Key","request_count":399,"valid_until":-1,"credits":-1,"enabled":true,"rate_limits":null,"rate_limits_enabled":false,"created_at":1693315042,"updated_at":1693484325},{"id":2,"api_key":"-Enyd-8kK2XThaaNkGZCFZXE9zBF-JFJ-0EYlY-UnQp_bTzRfXzhxp-YvQlEXwp5","api_key_hint":"...Xwp5","name":"New API Key","request_count":0,"valid_until":-1,"credits":-1,"enabled":true,"rate_limits":null,"rate_limits_enabled":false,"created_at":1693315040,"updated_at":1693315040},{"id":1,"api_key":"Z-lqb8MKAxEnQbwpY-4ttmOIQ8qHx7PcHSY0JbSaJNr9izyUJbocYvhSJVHJ3KTF","api_key_hint":"...3KTF","name":"New API Key","request_count":1273,"valid_until":-1,"credits":-1,"enabled":true,"rate_limits":null,"rate_limits_enabled":false,"created_at":1693285613,"updated_at":1693306502}]
```

**Comment by BitVoyagerMan** [comment]
@jacoblee93  Everything is available to public without a cost so that anyone can experience power of decentralized AI. It's fully open-source and its in test domain because UI needs to be added in there.

**Comment by jacoblee93** [comment]
> @jacoblee93 Everything is available to public without a cost so that anyone can experience power of decentralized AI. It's fully open-source and its in test domain because UI needs to be added in there.

It won't be forever though right?

I guess we could put this under experimental?

**Comment by BitVoyagerMan** [comment]
Yes, you're right. We will update the link in a week and this PR. Thanks for your effort regarding this.

**Comment by BitVoyagerMan** [comment]
> > @jacoblee93 Everything is available to public without a cost so that anyone can experience power of decentralized AI. It's fully open-source and its in test domain because UI needs to be added in there.
> 
> It won't be forever though right?
> 
> I guess we could put this under experimental?

@jacoblee93 We will update asap. Now test link is OK for experiment. Of course it wouldn't be forever.

**Comment by BitVoyagerMan** [comment]
@jacoblee93  Can you merge this PR asap? Because this PR is so important for us to move forward. We will update url 2 or 3 weeks later. But if you can, please review and merge this PR. Thanks.

**Comment by jacoblee93** [comment]
Can you move this under `langchain/experimental`?

**Comment by BitVoyagerMan** [comment]
> Can you move this under `langchain/experimental`?

@jacoblee93  Sorry.  We have to integrate into llm. Already integrated to langchain python library with the same code.
We update the url asap. The reason we want integrate this right now inspite of test url, we have to move forward to flowiseAI integration using BittensorLLM which is integrated into llm of langchain-js.

**Comment by jacoblee93** [comment]
No I mean under `langchain/experimental/llms`, similarly to: https://github.com/hwchase17/langchainjs/blob/main/langchain/src/experimental/chat_models/anthropic_functions.ts

**Comment by BitVoyagerMan** [comment]
Totally agree.

**Pull request closed without merging**